lib/ansible/modules/cloud/docker/docker_plugin.py

```diff
@@ -0,0 +1,297 @@
+#!/usr/bin/python
+# coding: utf-8
+#
+# Copyright: (c) 2019, Vladimir Porshkevich (@porshkevich) <neosonic@mail.ru>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = u'''
+module: docker_plugin
+version_added: "2.8"
+short_description: Manage Docker plugins
+description:
+  - Install/remove Docker plugins.
+  - Performs largely the same function as the "docker plugin" CLI subcommand.
+options:
+  name:
+    description:
+      - Name of the plugin to operate on.
+    required: true
+    type: str
+
+  alias:
+    description:
+      - Alias of the plugin to operate on. Same plugin can be installed with different alias.
+    type: str
+
+  plugin_options:
+    description:
+      - Dictionary of plugin settings.
+    type: dict
+
+  state:
+    description:
+      - C(absent) remove the plugin.
+      - C(present) install the plugin, if it does not already exist.
+      - C(enable) enable the plugin.
+      - C(disable) disable the plugin.
+    default: present
+    choices:
+      - absent
+      - present
+      - enable
+      - disable
+
+extends_documentation_fragment:
+  - docker
+  - docker.docker_py_2_documentation
+
+author:
+  - Vladimir Porshkevich (@porshkevich)
+
+requirements:
+  - "python >= 2.7"
+  - "docker-py >= 2.6.0"
+  - "Please note that the L(docker-py,https://pypi.org/project/docker-py/) Python
+    module has been superseded by L(docker,https://pypi.org/project/docker/)
+    (see L(here,https://github.com/docker/docker-py/issues/1310) for details).
+    Version 2.1.0 or newer is only available with the C(docker) module."
+  - "Docker API >= 1.25"
+'''
+
+EXAMPLES = '''
+- name: Install a plugin
+  docker_plugin:
+    name: plugin_one
+
+- name: Remove a plugin
+  docker_plugin:
+    name: plugin_one
+    state: absent
+
+- name: Install a plugin with options
+  docker_plugin:
+    name: weaveworks/net-plugin:latest_release
+    alias: weave-net-plugin
+    plugin_options:
+      IPALLOC_RANGE: "10.32.0.0/12"
+      WEAVE_PASSWORD: XXXXXXXX
+'''
+
+RETURN = '''
+facts:
+    description: Plugin inspection results for the affected plugin.
+    returned: success
+    type: dict
+    sample: {}
```

Comment:
I would return an explicit boolean flag which informs whether the plugin is activated or not (for `state != 'absent'`).